wallet/src/main/java/com/android/identity_credential/wallet/MainActivity.kt

```diff
@@ -0,0 +1,477 @@
+                    MainScreenNoCredentialsAvailable(navigation)
+                } else {
+                    MainScreenCredentialPager(navigation)
+                    Row(
+                        modifier = Modifier.fillMaxWidth(),
+                        horizontalArrangement = Arrangement.Center
+                    ) {
+                        Text(
+                            modifier = Modifier.padding(8.dp),
+                            text = "Hold to Reader"
+                        )
+                    }
+                }
+            }
+        }
+    }
+
+    @Composable
+    fun MainScreenNoCredentialsAvailable(navigation: NavHostController) {
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.Center
+        ) {
+            Text(
+                modifier = Modifier.padding(8.dp),
+                text = "No credentials in wallet, start by\n" +
+                "adding credentials.",
+                color = MaterialTheme.colorScheme.secondary,
+                textAlign = TextAlign.Center
+            )
+        }
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.Center
+        ) {
+            Button(onClick = {
+                navigation.navigate("AddToWalletScreen")
+            }) {
+                Text("Add to Wallet")
+            }
+        }
+    }
+
+    @Composable
+    fun MainScreenCredentialPager(navigation: NavHostController) {
+
+        Column() {
+
+            val credentialIds = application.credentialStore.listCredentials()
+            val pagerState = rememberPagerState(pageCount = {
+                credentialIds.size
+            })
+            HorizontalPager(
+                state = pagerState,
+                modifier = Modifier.height(200.dp)
+            ) { page ->
+
+                val credentialId = credentialIds[page]
+                val credential = application.credentialStore.lookupCredential(credentialId)!!
+                val encodedArtwork = credential.applicationData.getData("artwork")
+                val options = BitmapFactory.Options()
+                options.inMutable = true
+                val credentialBitmap =
+                    BitmapFactory.decodeByteArray(encodedArtwork, 0, encodedArtwork.size, options)
+                val credentialName = credential.applicationData.getString("displayName")
+
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.Center
+                ) {
+                    Image(
+                        bitmap = credentialBitmap.asImageBitmap(),
+                        contentDescription = "Artwork for $credentialName",
+                        modifier = Modifier.clickable(onClick = {
+                            navigation.navigate("CredentialInfo/$credentialId")
+                        })
```

Comment:
the card image looks really slick on-screen! I also like the HorizontalPager

Reply:
Cool, yea, I think we actually want to spend some time on getting some nice generic artwork and writing code to overlay parts of the holder's name.